pkg/s3select/csv/record.go

```diff
@@ -61,30 +61,28 @@ func (r *Record) Set(name string, value *sql.Value) error {
 	return nil
 }
 
-// MarshalCSV - encodes to CSV data.
-func (r *Record) MarshalCSV(fieldDelimiter rune) ([]byte, error) {
-	buf := new(bytes.Buffer)
-	w := csv.NewWriter(buf)
-	w.Comma = fieldDelimiter
-	if err := w.Write(r.csvRecord); err != nil {
-		return nil, err
+// WriteCSV - encodes to CSV data.
+func (r *Record) WriteCSV(w io.Writer, fieldDelimiter rune) error {
```

Comment:
You could accept `writer` argument instead of `w` which helps to avoid unpleasant name `csvW`. Also it would make minimum diffs to this method.

I think this comment is applicable to `WriteCSV` and `WriteJSON` all `pkg/s3select/*/record.go` sources.

Reply:
changed